### PR TITLE
Removed CookieRenderer by making it fully integrated into cookie render pass

### PR DESCRIPTION
**Pull request by mvaligursky**
- refactor on the way to make render passes more standalone, easier to schedule.

### CODE REVIEW

File: src/scene/renderer/render-pass-cookie-renderer.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
+import { Debug } from '../../core/debug.js';
```
Comment (willeastcott) [inline src/scene/renderer/render-pass-cookie-renderer.js:1]: What about a folder `renderer/render-passes`?

File: src/scene/renderer/render-pass-cookie-renderer.js [hunk 2]
```diff
@@ -74,6 +77,52 @@ class RenderPassCookieRenderer extends RenderPass {
...
+
+        // pick lights we need to update the cookies for
+        const filteredLights = this._filteredLights;
+        this.filter(lights, filteredLights);
+
+        // enabled / disable the pass
+        this.executeEnabled = filteredLights.length > 0;
+    }
+
+    filter(lights, filteredLights) {
```
Comment (willeastcott) [inline src/scene/renderer/render-pass-cookie-renderer.js:103]: I'm guessing this is much faster than `Array#filter`?

**Review by willeastcott — APPROVED** [review]

File: src/scene/renderer/render-pass-cookie-renderer.js (hunk 1 above)
Comment (mvaligursky) [reply src/scene/renderer/render-pass-cookie-renderer.js:1]: maybe if we have more of those, yeah

File: src/scene/renderer/render-pass-cookie-renderer.js (hunk 2 above)
Comment (mvaligursky) [reply src/scene/renderer/render-pass-cookie-renderer.js:103]: this avoids temporary array allocation, so why not

**Pull request merged**